### DISCUSSION TITLE
How to capture a still camera image (snapshot) to file from Eufy doorbell?

### DISCUSSION THREAD

**Discussion by ghbirb** [fuatakgun/eufy_security#739] [Q&A]
Hi,
I am new to Home Assistant and I am currently learning the ropes.

I have installed HACS, Eufy Security, RTSP and the Eufy Security add-on (version showing 1.3.0 – I do not think this is the latest version however as the GitHub page shows v6.2?). 

All appears to be working so far fine.

I am attempting to display a notification when the doorbell is pressed on my TV with a still image from a Eufy Dual CAM 2K doorbell (battery) using the **camera.snapshot** service.

I have everything working including the button press from the doorbell as a trigger, the TV notification etc however **camera.snapshot** does not generate an image file – it does nothing. I have the directory defined in config.yaml using *allowlist_external_dirs*.

I assume the **camera.snapshot** service is unsupported by the Eufy Security add-on or something more needs to be done? 

How do I get a still current photo at the time of a button press from a Eufy doorbell?

**Comment by P00DLEMAN** [comment]
Same issue here with 2K battery doorbell T8210. The doorbell is P2P only and I have not been able to get any live feed or last event screenshot. Currently attempting the below automations to start and end the stream, but nothing happens.

alias: Doorbell Picture Update
description: ""
trigger:
  - platform: state
    entity_id:
      - binary_sensor.doorbell_motion_detected
      - binary_sensor.doorbell_person_detected
    to: "on"
condition: []
action:
  - service: eufy_security.start_p2p_livestream
    data: {}
    target:
      entity_id: camera.doorbell_2
  - service: eufy_security.generate_image
    data: {}
    target:
      entity_id: camera.doorbell_2
  - service: notify.mobile_app_ian_s_iphone
    data:
      message: Someone is at the front door
      data:
        image: /api/camera_proxy/camera.doorbell_2
mode: single

alias: Doorbell Stream Stop
description: ""
trigger:
  - platform: state
    entity_id:
      - binary_sensor.doorbell_motion_detected
      - binary_sensor.doorbell_person_detected
    to: "off"
condition:
  - condition: state
    entity_id:
      - binary_sensor.doorbell_motion_detected
      - binary_sensor.doorbell_person_detected
    state: "off"
action:
  - service: eufy_security.stop_p2p_livestream
    data: {}
    target:
      entity_id: camera.doorbell_2
mode: single

Dashboard:

      - type: conditional
        conditions:
          - entity: camera.doorbell_2
            state: idle
        card:
          show_state: true
          show_name: true
          camera_view: auto
          type: picture-entity
          entity: camera.doorbell_2
          camera_image: camera.doorbell_2
          tap_action:
            action: none
          hold_action:
            action: none
      - type: conditional
        conditions:
          - entity: camera.doorbell_2
            state: streaming
        card:
          type: vertical-stack
          cards:
            - type: custom:webrtc-camera
              entity: camera.doorbell_2

I have not been able to get the camera entity to change from "Idle". Any ideas?

**Comment by ghbirb** [comment]
Any ideas with this issue?
